### PR TITLE
Georgi Shopov - HW07

### PR DESCRIPTION
**Pull request by georgiiishopovv**

### CODE REVIEW

**Comment by kajigor** [comment]
2 ok

File: monads/monads/test/Spec.hs [hunk 1]
```diff
@@ -0,0 +1,113 @@
...
+properExecution = property $ do
+    expr <- forAll (genExpr)
+    let program = compile expr
+    let initial = MachineState [] (M.fromList [("x", 10), ("y", 20), ("z", 30), ("a", 8), ("b", 33)]) 
+    let result = execProgram program initial
+    case result of
+        Right executed -> do
+            let stack = getStack executed
+            case stack of
+                [one] -> success
```
Comment (kajigor) [inline monads/monads/test/Spec.hs]: You should also check that the result is correct.

File: monads/monads/test/Spec.hs [hunk 2]
```diff
@@ -0,0 +1,113 @@
...
+    let initial = MachineState [] (M.fromList [("x", 10), ("y", 20), ("z", 30), ("a", 8), ("b", 33)])
+    let result = execProgram program initial
+    case result of
+        Left (StackUnderflow _) -> success
+        _ -> fail "StackUnderflow Error Undetected"
+
+-- Undefined var error
+undefinedVarError :: Property
+undefinedVarError = property $ do
+    x <- forAll (genVarExpr)
```
Comment (kajigor) [inline monads/monads/test/Spec.hs]: You can also define this property for a properly random expression.

**Review by kajigor — CHANGES_REQUESTED** [review]

File: monads/monads/test/Spec.hs (hunk 1 above)
Comment (georgiiishopovv) [reply monads/monads/test/Spec.hs]: fixed

File: monads/monads/test/Spec.hs (hunk 2 above)
Comment (georgiiishopovv) [reply monads/monads/test/Spec.hs]: fixed

**Comment by kajigor** [comment]
Ok

**Pull request closed without merging**